src/provisioner/validators.py

```diff
@@ -148,29 +147,60 @@ def _check_number_of_nodes(self):
     def _check_external_services(self):
         """" Validate no of external services in cortx config """
 
-        sol_conf_ext_services = self._get_value_from_solution_config(
+        cortx_config_ext_services = self._get_config(
             ConfigValidator._key_ext_service)
         conf_stor_ext_services = self._get_value_from_conf_store(
             ConfigValidator._key_ext_service)
 
-        if len(conf_stor_ext_services) != len(sol_conf_ext_services):
+        if len(conf_stor_ext_services) != len(cortx_config_ext_services):
             raise CortxProvisionerError(errno.EINVAL,
                 'External services define in conf_store is not same as '
-                f'solution_config. External services in {self.solution_conf_url}'
-                f' is {len(sol_conf_ext_services)} and in {self.cortx_conf_url}'
+                f'cortx config. External services in {self.solution_conf_url}'
+                f' is {len(cortx_config_ext_services)} and in {self.cortx_conf_url}'
                 f' is {len(conf_stor_ext_services)}')
 
         # Check endpoints
         for service in conf_stor_ext_services.keys():
             if (conf_stor_ext_services[service]['endpoints'] !=
-                    sol_conf_ext_services[service]['endpoints']):
+                    cortx_config_ext_services[service]['endpoints']):
                 raise CortxProvisionerError(errno.EINVAL,
                     f'{service} endpoints define in conf_store is not same as '
-                    'solution_config.')
+                    'cortx config.')
+        return 0
+
+    def _validate_components(self):
+        """Verify components defined in cluster.yaml is supported in constant file."""
+        node_types = self._get_config('cluster>node_types')
```

Comment:
This one will not work with consul based conf store. Need to get exact key.

Reply:
This key I am reading from cluster.yaml file and not from the confstore.